### PR TITLE
Chapter4 5 6

### PR DESCRIPTION
**Pull request by SergioGueruiz**
Hello!
This PR contains all i saw in chapter 4, 5 and 6
Loops with while and until
Condiotionals if, elsif, else and unless
Operators | |, && and !

### CODE REVIEW

File: Chapter4/string.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+puts 'Juliet said "Goodbye" to Romeo'#prints double quotes
+
+puts "Juliet said \"Goodbye\" \tto Romeo"
+puts 'Juliet said \'Goodbye\' to Romeo'
+#If we want put quotes in our text we can put \ before each quote
+
+#\n new line
+#\t tab
+
+#Single quotes vs Double quotes
```
Comment (mauriciosabogal) [inline Chapter4/string.rb:37]: What is the difference between `Single Qoutes` and `Double Qoutes`? 
What cases should you use single quotes? 
What cases should you use double quotes?

File: Chapter5/def.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+#Default or optional parameters
+
+def makePhoneCall(number, internationalCode = "0", area_code = "000")
+  return "Calling #{internationalCode}-#{area_code}-#{number}"
+end
+#The default parameters must be at the final
+p makePhoneCall("123456") #One argument, and takes the other by default
+p makePhoneCall("12347", 2) #Two agruments, replaces the value and takes the other by default
+p makePhoneCall(1235, 7, 111) #Three arguments, replaces the values by default
```
Comment (mauriciosabogal) [inline Chapter5/def.rb:57]: What happens if you don't pass any arguments to this `makePhoneCall` function?

File: Chapter6/caseStatement.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#Case statement
+#This can replace many elsif
+def rateFood(food)
+  case food
```
Comment (mauriciosabogal) [inline Chapter6/caseStatement.rb:4]: When do you think you should use `case` instead of `if/else`?

File: Chapter6/fizzBuzz.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+def fizzBuzz(num)
```
Comment (mauriciosabogal) [inline Chapter6/fizzBuzz.rb:1]: Can you explain what this function is doing?

File: Chapter6/keyWord.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
+#Unless keyword
+
+pass = "abcd"
+#This is the opposite to if statement
+unless pass == "topsecret"
```
Comment (mauriciosabogal) [inline Chapter6/keyWord.rb:5]: When should you use `unless` instead of `if`?

File: Chapter5/def.rb (hunk 2 above)
Comment (SergioGueruiz) [reply Chapter5/def.rb:57]: Shows an **error** and don't run the next lines, is necessary at least one argument to run the function

File: Chapter6/fizzBuzz.rb (hunk 4 above)
Comment (SergioGueruiz) [reply Chapter6/fizzBuzz.rb:1]: Gets a number and creates a var `count`, the var `num` is the limit to `count`, in the loop are many condition if `count` meets with any condition prints something like a fiiz, buzz or fizzbuzz, but if `count` doesn't meets prints the value of `count` at the end of each loop adds one to `count` and starts the loop again.

File: Chapter4/string.rb (hunk 1 above)
Comment (SergioGueruiz) [reply Chapter4/string.rb:37]: -If we want to do Interpolation String or adds a new line `\n` (Escape) we should use `Double Quotes` , 'Cause `Single quotes` don't support them.
-If we only need shows text we can use `Single quotes`
-If we need use Interpolation String or Escape characters we must use `Double Quotes`

File: Chapter6/caseStatement.rb (hunk 3 above)
Comment (SergioGueruiz) [reply Chapter6/caseStatement.rb:4]: When i have to do many conditons is better use `case` than `if/else`

File: Chapter6/keyWord.rb (hunk 5 above)
Comment (SergioGueruiz) [reply Chapter6/keyWord.rb:5]: When i want comprobate if something is `false`,  instead of using `if` with `!` operator

**Review by mauriciosabogal — APPROVED** [review]

**Pull request merged**